### PR TITLE
Ballot graph overhaul

### PR DESCRIPTION
**Pull request by cdonnay**
Sorry for the length and number of changes here, but I was finding the BallotGraph object to not really be working as intended. High level summary of changes:

1. Change `complete` parameter to `allow_partial` in `__init__`. If `allow_partial` is True, BallotGraph constructs all possible ballot types. If False, it only generates full linear orderings.
2. Add `fix_short` parameter to `__init__`. Gives the user the option to change ballots of length n-1 to n.
3. Add several parameters to `draw`. `to_display` lets the user specify a boolean function on nodes to be drawn. `neighborhoods` and `show_cast` are parameters that give common display functions.
4. Added/replaced necessary test functions.
5. Changed node_data to node_weights since that is the only kind of data it was storing.

@jamesturk @drdeford @jgibson517 @jennjwang @ziglaser

### CODE REVIEW

File: src/votekit/graphs/ballot_graph.py [hunk 1]
```diff
@@ -185,9 +190,13 @@ def fix_short_ballot(self, ballot: list, candidates: list) -> list:
 
         return ballot + list(missing)
 
-    def label_cands(self, candidates):
+    def label_cands(self, candidates,
+                    to_display: Optional[Callable] = all_nodes):
```
Comment (jgibson517) [inline src/votekit/graphs/ballot_graph.py:194]: Here's the issue with the type hinting as there's technically None being passed when this argument is not included. Changing it to `to_display: Optional[Union[Callable, None]]` I think will fix the linting issue. 

Separately, can you explain why this is a callable, what is the user supposed to input if they want to to only display a subset of the nodes? Is it a function that returns True/False for a given node?

Comment (cdonnay) [reply src/votekit/graphs/ballot_graph.py]: Yeah, the callable should take in the graph and a node and return True if that node should be displayed. I just added an edit to make sure the graph was an argument to these callables.

Comment (cdonnay) [reply src/votekit/graphs/ballot_graph.py]: Ah looks like the thing that fixed it was just using `Callable`. Now the only thing failing is `test_interval_from_params`, which is that non-deterministic test. Can you re-run it for me @jgibson517 ? Thanks!

**Comment by cdonnay** [comment]
Updated the draw method to ensure labels do not go over the edge of the plot.

File: src/votekit/graphs/ballot_graph.py (hunk 1 above)
Comment (jgibson517) [reply src/votekit/graphs/ballot_graph.py]: 👍

**Pull request merged**